charmcraft.yaml

```diff
@@ -90,11 +89,275 @@ config:
+    secret-persistence:
+      description: |
+        Defines the Secret Persistence type. Defaults to NO_OP.
+
+        Acceptable values are: "GOOGLE_SECRET_MANAGER", "AWS_SECRET_MANAGER", "TESTING_CONFIG_DB_TABLE", "VAULT"
+      type: string
+
+    secret-store-gcp-project-id:
+      description: Defines the GCP Project to store secrets in. Alpha support.
+      type: string
+
+    secret-store-gcp-credentials:
+      description: |
+        Defines the JSON credentials used to read/write Airbyte Configuration to Google Secret Manager.
+        
+        These credentials must have Secret Manager Read/Write access. Alpha support.
+      type: string
+
+    vault-address:
+      description: Defines the vault address to read/write Airbyte Configuration to Hashicorp Vault. Alpha Support.
+      type: string
+
+    vault-prefix:
+      description: Defines the vault path prefix. Empty by default. Alpha Support.
+      type: string
+
+    vault-auth-token:
+      description: The token used for vault authentication. Alpha Support.
+      type: string
+
+    vault-auth-method:
+      description: How vault will perform authentication. Currently, only supports Token auth. Defaults to "token". Alpha Support.
+      default: "token"
+      type: string
+
+    aws-access-key:
+      description: Defines the aws_access_key_id from the AWS credentials to use for AWS Secret Manager.
+      type: string
+
+    aws-secret-access-key:
+      description: Defines aws_secret_access_key to use for the AWS Secret Manager.
+      type: string
+
+    aws-kms-key-arn:
+      description: Optional param that defines the KMS Encryption key used for the AWS Secret Manager.
+      type: string
+
+    aws-secret-manager-secret-tags:
+      description: |
+        Defines the tags that will be included to all writes to the AWS Secret Manager.
+        
+        The format should be "key1=value1,key2=value2".
+      type: string
+
+    ##### Jobs config #####
+    sync-job-retries-complete-failures-max-successive:
+      description: Max number of successive attempts in which no data was synchronized before failing the job.
+      default: 5
+      type: int
+
+    sync-job-retries-complete-failures-max-total:
+      description: Max number of attempts in which no data was synchronized before failing the job.
+      default: 10
+      type: int
+
+    sync-job-retries-complete-failures-backoff-min-interval-s:
+      description: Minimum backoff interval in seconds between failed attempts in which no data was synchronized.
+      default: 10
+      type: int
+
+    sync-job-retries-complete-failures-backoff-max-interval-s:
+      description: Maximum backoff interval in seconds between failed attempts in which no data was synchronized.
+      default: 1800
+      type: int
+
+    sync-job-retries-complete-failures-backoff-base:
+      description: Exponential base of the backoff interval between failed attempts in which no data was synchronized.
```

Comment:
Does this one also need a unit in the name?

Reply:
No units needed here since this is a backoff exponent for the retries. With the default value of 3, it would first retry after 10 seconds, then 30 seconds, then 90 seconds etc.